lib/cloud/clients.go

```diff
@@ -62,9 +64,12 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 
+	apiawsutils "github.com/gravitational/teleport/api/utils/aws"
 	libcloudaws "github.com/gravitational/teleport/lib/cloud/aws"
 	"github.com/gravitational/teleport/lib/cloud/azure"
 	"github.com/gravitational/teleport/lib/cloud/gcp"
+	"github.com/gravitational/teleport/lib/services"
```

Comment:
nit: if possible, we should avoid pulling `lib/services` to `lib/cloud`

Reply:
hmm, it's there for `services.AssumeRole`. I could move that somewhere else, or refactor. I've tried to make it easy for cloud clients callers to use the functions without doing a lot of checking and type conversions for what they pass in. I suppose I could change the signature to `externalID string, roleARNs ...string`, since there is only one source of external ID really?